### PR TITLE
Update, maintain, and optimize extension

### PR DESCRIPTION
**Pull request by jecassis**
- Update VS Code engine to 1.56.0 to support Node.js 14
 - Update several dependencies to address CVEs
 - Update ESLint to version 8 and implement some suggested fixes
 - Update Prettier and apply suggested formatting changes
 - Remove React-related ESLint dependencies
 - Remove direct dependency on `antlr4` package as it is a submodule of `antlr4ts`
 - Remove on Markdown activation
 - Rename `.eslintrc` to `.eslintrc.json` to fix ESLint editor integration
 - Add `-Xexact-output-dir` to compile command to fix MacOS build
 - Update `README.md` with a suggested configuration section and add missing settings
 - Enable bracket matching and coloring to function with SystemVerilog textual brackets
 - Fix `port-net-parameter` highlighting corner case
 - Add Github Actions test workflow
 - Fix coverage collection

### CODE REVIEW

**Comment by eirikpre** [comment]
Joe made me realize that I'm not able to follow well enough on my own.
You (@jecassis and @joecrop) are invited to be collaborators if you want, so that you do not need to wait for me.

I think we should wait until the other pull-requests are merged before this one is.

There is no automated delivery process from the GitHub repo to MS Marketplace and for the code updates to have effect we need to publish to marketplace: [ref](https://code.visualstudio.com/api/working-with-extensions/publishing-extension)
I will look into a way to automate it or give you publisher rights.

**Comment by joecrop** [comment]
@eirikpre I would be delighted to be a collaborator. Please set up the permissions for write access on your end. 

Also, @jecassis I am happy to help review your PRs like this one if you want to assign me as a reviewer. Are you interested in reviewing any of my PRs?

**Comment by eirikpre** [comment]
@joecrop You should have write(push) access now?

I tried look for why you wouldn't be able to, but seems like setting permissions explicitly is only available to organizations.
This is how it looks for me:  https://stackoverflow.com/a/55135314/9084781

**Comment by joecrop** [comment]
Hi @jecassis, I finally merged in the major change to the reference provider. Can you please rebase and apply your changes on top of the latest release?

**Comment by jecassis** [comment]
@joecrop Merged from master branch as requested, in addition to the PR changes prior to the merge:
- Resolved the merge conflicts and issues; particularly with ESLint.
- Updated `CHANGELOG.md`; mostly with the PR comments; feel free to edit/modify to your liking.
- Bumped the version to `0.12.1`; also feel free to change to suit your preferred versioning scheme.
- Updated a lot of NPM packages to their latest versions; did not update those that included breaking changes.
- Added caching and other goodies to the GitHub Actions pipeline.
- Increased the minimum version of VS Code to 1.61.0. This is the version that NPM picks up for the TypeScript types so matched that so that `vsce` does not throw an error during packaging/publishing.

Thank you for circling back to this PR after all the nice improvements.

File: README.md
Comment (joecrop) [inline README.md:55]:
```suggestion
- `systemverilog.enableIncrementalIndexing`: _Boolean_, Enable incremental indexation as files are opened
```

File: src/compiling/DocumentCompiler.ts
Comment (joecrop) [inline src/compiling/DocumentCompiler.ts:169]: Can we delete this line instead of waiving it?

File: src/indexer.ts
Comment (joecrop) [inline src/indexer.ts:98]: I think the `async` should be removed from this function call instead of being waived.

File: src/indexer.ts
Comment (joecrop) [inline src/indexer.ts:230]: I think the `async` should be removed from this function call instead of being waived.

**Review by joecrop — CHANGES_REQUESTED** [review]

File: src/indexer.ts
Comment (jecassis) [reply src/indexer.ts:98]: When I remove this `async` (and the corresponding `await` below) some indexer tests fail. Since I don't know the exact intent or usage model, I left it as is.

File: src/indexer.ts
Comment (jecassis) [reply src/indexer.ts:230]: When I remove this async one indexer test fails. Left it as is.

**Review by joecrop — APPROVED** [review]

**Pull request merged**
